config/settings.rb

```diff
@@ -29,6 +29,9 @@
   SETTINGS[:fqdn] ||= Facter.value(:fqdn)
 end
 
+SETTINGS[:hosts] = [] unless SETTINGS[:hosts].is_a?(Array)
```

Comment:
This has a big risk of masking misconfigurations. Let's say the user has the following settings:
```yaml
:hosts: my.fdqn.example.com
```

This results in it being silently ignored, which is IMHO always a bad thing when it comes to configuration.

A more common approach is:
```suggestion
SETTINGS[:hosts] ||= []
```

Then the later code may fail, but at least it'll be clear since the application doesn't start up.